numba/parfor.py

```diff
@@ -1371,8 +1330,8 @@ def parfor_insert_dels(parfor, curr_dead_set):
         l.step.name for l in parfor.loop_nests if isinstance(
             l.step, ir.Var)}
     loop_vars |= {l.index_variable.name for l in parfor.loop_nests}
-    for var_list in parfor.array_analysis.array_size_vars.values():
-        loop_vars |= {v.name for v in var_list if isinstance(v, ir.Var)}
+    # for var_list in parfor.array_analysis.array_size_vars.values():
```

Comment:
I think this commented code can be removed. gufunc parfor lowering used to rely on array size variables to stay alive, but not anymore.